### PR TITLE
GuestConfiguration CLI initial checkin

### PR DESCRIPTION
**Pull request by gehuang-dut**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [x] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
GuestConfiguration

**Comment by gehuang-dut** [comment]
The style check error is caused by: 

"Running flake8 on CI scripts...
The following directories are missing codeowners ['/src/guestconfig/'].
Pylint OK.
Other Static checks...
Errors whilst verifying /home/vsts/work/1/s/.github/CODEOWNERS!"

I don't have permission to edit the .github file. Other checks are all green. Thanks.

**Comment by changlong-liu** [comment]
Hi @gehuang-dut , to me the CODEOWNERS is just a normal file: https://github.com/Azure/azure-cli-extensions/blob/master/.github/CODEOWNERS

Could you modify and commit it?

**Comment by gehuang-dut** [comment]
> 
> 
> Hi @gehuang-dut , to me the CODEOWNERS is just a normal file: https://github.com/Azure/azure-cli-extensions/blob/master/.github/CODEOWNERS
> 
> Could you modify and commit it?

Ok, it's a mistake at my side. Fixed. Green now. Thanks.

File: src/guestconfig/README.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+Microsoft Azure CLI 'guestconfig' Extension
```
[the remark this answers is not in the document]
Comment (gehuang-dut) [reply src/guestconfig/README.md:1]: updated. Thanks!

File: src/guestconfig/azext_guestconfig/tests/latest/preparers.py [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+from azure.cli.testsdk.reverse_dependency import get_dummy_cli
+
+
+KEY_RESOURCE_GROUP = 'rg'
+KEY_VIRTUAL_NETWORK = 'vnet'
+KEY_VNET_SUBNET = 'subnet'
+KEY_VNET_NIC = 'nic'
+
+
+class VirtualNetworkPreparer(NoTrafficRecordingPreparer, SingleValueReplacer):
```
Comment (fengzhou-msft) [inline src/guestconfig/azext_guestconfig/tests/latest/preparers.py:25]: This file can be deleted as `VirtualNetworkPreparer` is already moved into `azure-cli-testsdk` and it's actually not used in the tests for this extension.

Comment (gehuang-dut) [reply src/guestconfig/azext_guestconfig/tests/latest/preparers.py]: > 
> 
> This file can be deleted as `VirtualNetworkPreparer` is already moved into `azure-cli-testsdk` and it's actually not used in the tests for this extension.

removed. Thanks !

Comment (gehuang-dut) [reply src/guestconfig/azext_guestconfig/tests/latest/preparers.py]: > 
> 
> This file can be deleted as `VirtualNetworkPreparer` is already moved into `azure-cli-testsdk` and it's actually not used in the tests for this extension.

rebased from master and it's green. Thanks.

**Pull request merged**